index.js

```diff
@@ -78,6 +83,196 @@ function loader (version) {
         return new Item(item.blockId, item.itemCount, item.itemDamage, item.nbtData)
       }
     }
+
+    static getEnchants (item) {
+      const normalize = (enchList) => enchList.map(ench => {
+        return { lvl: ench.lvl, name: mcData.enchantments[ench.id].name }
+      })
+      if (mcData.isOlderThan('1.13')) {
+        let itemEnch
+        if (item.name === 'enchanted_book' && item.nbt !== null) {
+          itemEnch = nbt.simplify(item.nbt).StoredEnchantments
+        } else if (item.nbt !== null) {
+          itemEnch = nbt.simplify(item.nbt).ench
+        } else {
+          itemEnch = []
+        }
+        return normalize(itemEnch)
+      }
+      const gg = nbt.simplify(item.nbt)
+      console.log(gg)
+      return 1
+    }
+
+    // denormalize converts normalized enchants back to 1.8 enchant format with {id, lvl}
+    static denormalize (enchantList) {
+      const findEnch = (name) => Object.entries(mcData.enchantments).map(x => x[1]).find(x => x.name === name)
+      return enchantList.map(ench => ({ lvl: ench.lvl, id: findEnch(ench.name).id }))
+    }
+
+    static anvil (itemOne, itemTwo, creative, rename) {
+      function baseCost (item) {
+        if (item.name === 'enchanted_book') return 0
+
+        return item?.nbt?.value?.RepairCost?.value ?? 0
+      }
+
+      function renameCost (item) {
+        if (item?.nbt?.value?.RepairCost?.value === 0x7fffffff) {
+          throw new Error('Renaming impossible')
+        }
+        return 1
+      }
+
+      function combinePossible (itemOne, itemTwo) {
+        if (itemOne.name === 'enchanted_book' && itemTwo.displayName !== 'Enchanted Book') throw new Error('Can only combine book with book')
+        else if (itemOne.name === 'enchanted_book') return // return here because this func will throw because enchanted book isnt fixable
+        let [, fixMaterials] = canFixData.find(([items]) => items.includes(itemOne.displayName))
+        fixMaterials = fixMaterials.concat(['Enchanted Book', itemOne.displayName])
+        // if (itemOne.name === 'enchanted_book' && itemTwo.displayName !== 'Enchanted Book') throw new Error('Can only combine book with book')
+        if (!fixMaterials.includes(itemTwo.displayName)) throw new Error('Not able to be combined')
```

Comment:
This logic seems messed up, are you testing multiple time for the same thing ? This surely could be simplified.

Reply:
cleaned it up a bit

Reply:
done